### PR TITLE
PHC-4753 Add circle discussions screen

### PR DESCRIPTION
**Pull request by bruce-glazier**
## Changes

  - Fix baseURL issue with the GraphQL client (first commit)
  - Adds hook to auto-join available circles
  - Adds a very basic circle discussion screen, text only (no reactions, no replies, no media)

## Next PR(s)
- Add loading indicator for the loading scrollView
- Apply any feedback from this PR
- Add unit tests
- Possible renaming of components/hooks for consistency
- Identify cause of warnings shown in video

## Screenshots

https://user-images.githubusercontent.com/76954025/235528416-782c794c-2710-4ef4-8bcc-e2483463bb55.mp4

### CODE REVIEW

File: src/hooks/useJoinCircles.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  const { data } = useAppConfig();
+  const { accountHeaders } = useActiveAccount();
+  const { activeProject } = useActiveProject();
+  const { httpClient } = useHttpClient();
+
+  return useQuery(
+    `/v1/life-research/projects/${activeProject?.id}/app-config/circles`,
+    () => {
+      if (data?.homeTab?.circleTiles?.some((c) => !c.isMember)) {
+        console.log('And here?');
```
Comment (bruce-glazier) [inline src/hooks/useJoinCircles.tsx:17]: Remove....

Comment (bruce-glazier) [reply src/hooks/useJoinCircles.tsx]: Done

File: src/hooks/useGraphQLClient.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const GraphQLClientContextProvider = ({
     injectedGraphQLInstance || defaultGraphQLClientInstance;
 
   if (baseURL) {
-    graphQLInstance.setEndpoint(baseURL || defaultBaseURL);
+    graphQLInstance.setEndpoint(`${baseURL}/v1/graphql` || defaultBaseURL);
```
Comment (markdlv) [inline src/hooks/useGraphQLClient.tsx:41]: Just seeing this, but it seems like the `|| defaultBaseURL` here could never have been used.  It's just more glaring now

**Review by markdlv — DISMISSED** [review]
Nice!  Two questions to make sure I'm tracking:

- Is the idea to add unit tests for these components/hooks as they get fleshed out more?  Makes sense if so
- Is the idea that the PostsScreen (or perhaps just PostsList) will someday be embedded into a broader screen concept (e.g. CircleScreen) in future PRs?

File: src/hooks/useGraphQLClient.tsx (hunk 2 above)
Comment (bruce-glazier) [reply src/hooks/useGraphQLClient.tsx:41]: Oh yeah, 100%, I see that now.

**Comment by bruce-glazier** [comment]
> Nice! Two questions to make sure I'm tracking:
> 
> * Is the idea to add unit tests for these components/hooks as they get fleshed out more?  Makes sense if so

Yes, I plan to add unit tests in a follow up. I wanted to work out a few more things first/make the PRs a little smaller.


> * Is the idea that the PostsScreen (or perhaps just PostsList) will someday be embedded into a broader screen concept (e.g. CircleScreen) in future PRs?

Yes, I was thinking that the PostsList could encompass all the functionality so it may be added to a screen along side other components in the future. The naming, specifically PostsScreen, I don't think will ultimately be what we go with here. I may rename to be more "Circles" centric in another PR.

File: src/components/Circles/Post.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { formatRelative } from 'date-fns';
+import React, { useMemo } from 'react';
+import { View } from 'react-native';
+import { Avatar, Divider, List, Text } from 'react-native-paper';
+import { useStyles } from 'src/hooks';
```
[the remark this answers is not in the document]
Comment (bruce-glazier) [reply src/components/Circles/Post.tsx]: Done

File: src/hooks/useGraphQLClient.tsx (hunk 2 above)
Comment (bruce-glazier) [reply src/hooks/useGraphQLClient.tsx]: Fixed

**Review by markdlv — APPROVED** [review]

**Pull request merged**